### DISCUSSION TITLE
Scanning '/' fails occasionally with "no such file or directory"

### DISCUSSION THREAD

**Discussion by mudler** [aquasecurity/trivy#4299] [Bugs]
## Description

I'm observing a sporadic issue happening in our CI (almost 1/4 of our jobs fails with this now) since the latest trivy release (v0.38.2). We run trivy with:
```
trivy filesystem --format sarif -o report.sarif --no-progress /
```

## What did you expect to happen?

The scan to succeed

## What happened instead?

Occasionally, some scans fails with:

```
filesystem scan error: scan error: scan failed: failed analysis: walk filesystem: walk error: unknown error with /: unknown error with //tmp: unknown error with //tmp/rpm2110412680: no such file or directory
```

## Output of run with `-debug`:

We are using `trivy` in our CI pipelines, so we collected all the occurrences in here:
https://github.com/kairos-io/kairos/issues/1087

**Comment by mudler** [comment]
seems actually we are able to reproduce this also with trivy 0.37.3

**Comment by Itxaka** [comment]
This happens in the dir that trivy itself creates to store the Packages information when run in a rpm based system: https://github.com/aquasecurity/trivy/blob/main/pkg/fanal/analyzer/pkg/rpm/rpm.go#L83
